### PR TITLE
WT-8970 Check for positioned cursors before allowing RTS

### PR DESCRIPTION
**Pull request by keithbostic**

### CODE REVIEW

**Comment by keithbostic** [comment]
MDB patch build: https://evergreen.mongodb.com/version/6240a80056234336d30168ff?redirect_spruce_users=true

**Comment by keithbostic** [comment]
@luke-pearson, the MDB patch build looks clean, so I think this one is ready to go.

@lukech, this one has been a struggle to make work as you know, just flagging it for your attention.

File: src/txn/txn_rollback_to_stable.c [hunk 1]
```diff
@@ -1375,35 +1379,27 @@ __txn_user_active(WT_SESSION_IMPL *session)
...
+     * callers should be aware of this limitation.
      */
-    if (__txn_user_active(session)) {
+    if (cursor_active)
+        WT_RET_MSG(session, EBUSY, "rollback_to_stable illegal with active file cursors");
+    if (txn_active) {
+        ret = EBUSY;
         WT_TRET(__wt_verbose_dump_txn(session));
-        WT_RET_MSG(session, EBUSY, "rollback_to_stable illegal with active transactions");
+        WT_RET_MSG(session, ret, "rollback_to_stable illegal with active transactions");
```
Comment (luke-pearson) [inline src/txn/txn_rollback_to_stable.c:1400]: Why don't you just directly pass EBUSY here? Looks like it'll override any previously set ret. You could also `WT_IGNORE_RET` the verbose dump txn call.

File: src/txn/txn_rollback_to_stable.c [hunk 2]
```diff
@@ -1676,7 +1672,7 @@ __rollback_to_stable_btree_apply(
 
     if (perform_rts || max_durable_ts > rollback_timestamp || prepared_updates ||
       !durable_ts_found || has_txn_updates_gt_than_ckpt_snap) {
-        ret = __wt_session_get_dhandle(session, uri, NULL, NULL, 0);
+        ret = __wt_session_get_dhandle(session, uri, NULL, NULL, WT_DHANDLE_DISCARD);
```
Comment (luke-pearson) [inline src/txn/txn_rollback_to_stable.c:1675]: Can we get a comment explaining what the significance of `WT_DHANDLE_DISCARD` is here?

**Review by agorrod — APPROVED** [review]
This one slipped under my radar, but it seems like a great change. Thanks!

I didn't do a detailed review here, so don't rely just on this

File: src/txn/txn_rollback_to_stable.c (hunk 1 above)
Comment (keithbostic) [reply src/txn/txn_rollback_to_stable.c:1400]: Check out `WT_TRET`, generally the first error is retained and not overridden. (And, I wouldn't want EBUSY to override WT_PANIC, for example.) I wouldn't be too upset if `__wt_verbose_dump_txn()` stopped returning error values at all, but as long as it does return them, I don't like ignoring them, I think it's bad practice.

File: src/txn/txn_rollback_to_stable.c (hunk 2 above)
Comment (keithbostic) [reply src/txn/txn_rollback_to_stable.c:1679]: Fixed in 75c3444ba.

File: src/txn/txn_rollback_to_stable.c (hunk 1 above)
Comment (luke-pearson) [reply src/txn/txn_rollback_to_stable.c:1400]: So we'd be capturing `WT_PANIC` or other more severe errors than just `EBUSY`. I think that's fine.

**Review by luke-pearson — APPROVED** [review]
LGTM.

**Pull request merged**